index.js

```diff
@@ -4,140 +4,219 @@ window.addEventListener("load", () => {
+      firstNum = Number(currentString);
+      currentOp = "";
+      currentString = "";
+      clearCheck();
     } else {
       currentString = (Number(currentString) * -1).toString();
-      numDisplay.textContent = currentString;
+      numDisplay.textContent = rounder(Number(currentString));
+      firstNum = Number(currentString);
+      currentOp = "";
+      currentString = "";
+      clearCheck();
     }
   });
 
   for (let n = 0; n <= 3; n++) {
     opButtons[n].addEventListener("click", () => {
       if (errLock == true) {
         return;
+      } else if (firstNum != "" && currentOp != "" && currentString != "") {
+        equation();
+        currentOp = opArray[n];
+        clearCheck();
+      } else if (firstNum != "" && currentString == "") {
+        currentOp = opArray[n];
+        clearCheck();
       } else {
         firstNum = Number(currentString);
         currentOp = opArray[n];
         currentString = "";
-        numDisplay.textContent = currentString;
+        clearCheck();
       }
     });
   }
+
   opButtons[4].addEventListener("click", () => {
     if (errLock == true) {
       return;
-    } else if (currentString.includes("-")) {
-      currentString = "Error";
-      numDisplay.textContent = currentString;
+    } else if (
+      Number(currentString) < 0 ||
+      (currentString == "" && firstNum < 0)
+    ) {
+      numDisplay.textContent = "Error";
       firstNum = "";
       currentOp = "";
+      currentString = "";
       errLock = true;
+      clearCheck();
+    } else if (currentString == "" && firstNum != "") {
+      currentString = Math.sqrt(firstNum).toString();
+      numDisplay.textContent = rounder(Number(currentString));
+      firstNum = Number(currentString);
+      currentOp = "";
+      currentString = "";
+      clearCheck();
     } else {
-      currentString = Math.sqrt(Number(currentString))
-        .toString()
-        .substring(0, 10);
-      numDisplay.textContent = currentString;
+      currentString = Math.sqrt(Number(currentString)).toString();
+      numDisplay.textContent = rounder(Number(currentString));
+      firstNum = Number(currentString);
+      currentOp = "";
+      currentString = "";
+      clearCheck();
     }
   });
 
   opButtons[5].addEventListener("click", () => {
-    currentString = "";
-    numDisplay.textContent = currentString;
-    firstNum = "";
-    currentOp = "";
-    errLock = false;
+    if (clearState == "ac") {
+      numDisplay.textContent = "0";
+      firstNum = "";
+      currentOp = "";
+      currentString = "";
+      errLock = false;
+      clearCheck();
+    } else if (clearState == "c") {
+      numDisplay.textContent = "0";
+      currentString = "";
+      errLock = false;
+      clearCheck();
+    }
   });
 
   eqButton.addEventListener("click", () => {
     if (firstNum == "" && currentString == "") {
       return;
     } else if (firstNum == "" && currentString != "") {
       numDisplay.textContent = currentString;
```

Comment:
There's a bit of missing logic in here, possibly missing from my project description as well. If you type a number and press `=`, it should set that as firstNum and set currentString back to "". So if you type 1, 2, 3, =, 4, 5, it should display 45.

Reply:
I think I updated correctly? But please confirm.

Reply:
Seems right to me!